streaming/src/main/scala/org/apache/spark/streaming/receiver/ReceiverSupervisorImpl.scala

```diff
@@ -136,14 +136,18 @@ private[streaming] class ReceiverSupervisorImpl(
       metadataOption: Option[Any],
       blockIdOption: Option[StreamBlockId]
     ) {
+    var rBlock = receivedBlock
     val blockId = blockIdOption.getOrElse(nextBlockId)
     val numRecords = receivedBlock match {
       case ArrayBufferBlock(arrayBuffer) => arrayBuffer.size
+      case IteratorBlock(iterator) =>
+        var arrayBuffer = ArrayBuffer(iterator.toArray : _*)
```

Comment:
This is quite inefficient, as the data will be first copied to the arraybuffer, and then BlockManager will again use the ArrayBuffer's iterator to store the data. This is precisely why this wasnt done till now. :(